packages/core/src/fragments/IfcGeometryTiler/src/base-types.ts

```diff
@@ -1,30 +1,39 @@
-/**
- * A dictionary of geometries streamed from a server. Each geometry is identified by a unique number (id), and contains information about its bounding box, whether it has holes, and an optional file path for the geometry data.
- */
+interface GeometrySettings {
+  /** The bounding box of the geometry as a Float32Array. */
+  boundingBox: Float32Array;
+  /** A boolean indicating whether the geometry has holes. */
+  hasHoles: boolean;
+}
+
 export interface StreamedGeometries {
-  [id: number]: {
-    /** The bounding box of the geometry as a Float32Array. */
-    boundingBox: Float32Array;
-    /** A boolean indicating whether the geometry has holes. */
-    hasHoles: boolean;
-    /** An optional file path for the geometry data. */
-    geometryFile?: string;
-  };
+  [id: number]: GeometrySettings;
 }
 
 /**
  * A streamed asset, which consists of multiple geometries. Each geometry in the asset is identified by a unique number (geometryID), and contains information about its transformation and color.
  */
 export interface StreamedAsset {
-  /** The unique identifier of the asset. */
   id: number;
   /** An array of geometries associated with the asset. */
-  geometries: {
-    /** The unique identifier of the geometry. */
+  geometries: Array<{
     geometryID: number;
-    /** The transformation matrix of the geometry as a number array. */
+    /** The transformation matrix of the geometry. */
     transformation: number[];
-    /** The color of the geometry as a number array. */
     color: number[];
-  }[];
+  }>;
+}
+
+export type GeometryTileFileName = `${string}-${number}`;
+export type GlobalDataFileName = `${string}-global`;
+
+export interface GeometryTilesJson {
+  assets: Array<StreamedAsset>;
+  geometries: {
+    [id: number]: GeometrySettings & {
+      /** path to the geometry data file. Must have the index of the file as suffix. `-0` */
+      geometryFile: GeometryTileFileName;
```

Comment:
as `geometryFile` is only needed for typing the Json file, it is colocated here. The internal `GeometrySettings` type should never have this field. 

Its not a great idea to mix types by making fields optional, just because they share some other fields.